src/components/common/goodCategory/goodCategory.tsx

```diff
@@ -0,0 +1,41 @@
+import {Card} from "../card";
+import css from "./goods.module.css"
+import {Category, Good} from "../../../types";
+import {FC, useCallback, useEffect} from "react";
+import {Link} from "react-router-dom";
+import {useAppDispatch, useAppSelector} from "../../../store/hooks";
+import { getProductsSelector} from "../../../store";
+import {actionsCategories} from "../../../store/slices/sliceProducts";
+
+export const GoodCategory: FC<Category> = ({type, label, id}) => {
+
+    const dispatch = useAppDispatch()
+    const products = useAppSelector(getProductsSelector)
+
+    const returnProducts = useCallback(() => dispatch(actionsCategories.fetchProducts() as any), [dispatch])
```

Comment:
удал все из any. Ты починил типизацию хуком useAppDispatch, все должно быть ок
```suggestion
    const returnProducts = useCallback(() => dispatch(actionsCategories.fetchProducts()), [dispatch])
```